lib/backbeat/Metrics.js

```diff
@@ -301,6 +312,107 @@ class Metrics {
+            const numOfIntervals = Math.ceil(timeDisplay / INTERVAL);
+            const { requests } = res[0]; // Bytes done
+            let total = requests.slice(0, numOfIntervals)
+                .reduce((acc, i) => acc + i, 0);
+            // if timeDisplay !== OBJECT_MONITORING_EXPIRY, use internal timer
+            // and do not include the extra 4th interval
+            if (timeDisplay === EXPIRY) {
+                // all intervals apply, including 4th interval
+                const lastInterval =
+                    this._statsClient._normalizeTimestamp(new Date(now));
+                // in seconds
+                const diff = (now - lastInterval) / 1000;
+                // Get average for last interval depending on time passed so
+                // far for newest interval
+                total += ((INTERVAL - diff) / INTERVAL) *
+                    requests[numOfIntervals];
+            }
+            // Divide total by timeDisplay to determine data per second
+            const response = {
+                description: 'Current throughput for object replication in ' +
+                    'bytes/sec (throughput)',
+                throughput: (total / timeDisplay).toFixed(2),
+            };
+            return cb(null, response);
+        });
+    }
+
+    /**
+     * Get CRR progress for an object in bytes. Progress is the percentage of
+     * the object that has completed replication.
+     * @param {object} details - route details from lib/api/routes.js
+     * @param {function} cb - callback(error, data)
+     * @return {undefined}
+     */
+    getObjectProgress(details, cb) {
+        this._getData(details, undefined, (err, res) => {
+            if (err && err.type) {
+                this._logger.error('error getting metric: object progress', {
+                    origin: err.method,
+                    method: 'Metrics.getObjectProgress',
+                });
+                return cb(err.type.customizeDescription(err.message));
+            }
+            if (err || res.length !== details.dataPoints.length) {
+                this._logger.error('error getting metrics: object progress', {
+                    method: 'Metrics.getObjectProgress',
+                });
+                return cb(errors.InternalError);
+            }
+            // Find if time since start is less than OBJECT_MONITORING_EXPIRY
+            // time
+            const timeSinceStart = (Date.now() - this._internalStart) / 1000;
+            const timeDisplay = timeSinceStart < OBJECT_MONITORING_EXPIRY ?
+                timeSinceStart : OBJECT_MONITORING_EXPIRY;
+            const numOfIntervals = Math.ceil(timeDisplay / INTERVAL);
+            const [totalBytesToComplete, bytesComplete] = res.map(r => (
+                r.requests.slice(0, numOfIntervals).reduce((acc, i) =>
+                    acc + i, 0)
+            ));
+            let percentage = bytesComplete / totalBytesToComplete;
```

Comment:
May be better to check if `totalBytesToComplete === 0` than relying on `isNaN`, though not sure this case can actually happen there.

Also could be clearer with a separate var `const ratio = ...`, then `const percentage = ...` below.